compiler/src/iree/compiler/Codegen/Interfaces/BufferizationInterfaces.cpp

```diff
@@ -288,6 +290,62 @@ struct DispatchTensorStoreOpInterface
     return success();
   }
 };
+
+struct LoadFromMemrefOpInterface
+    : public BufferizableOpInterface::ExternalModel<
+          LoadFromMemrefOpInterface, IREE::Codegen::LoadFromMemrefOp> {
+  bool isWritable(Operation *op, Value value,
+                  const AnalysisState &state) const {
+    auto loadOp = cast<IREE::Codegen::LoadFromMemrefOp>(op);
+    return !loadOp.getReadOnly();
+  }
+
+  LogicalResult bufferize(Operation *op, RewriterBase &rewriter,
+                          const BufferizationOptions &options) const {
+    auto loadOp = cast<IREE::Codegen::LoadFromMemrefOp>(op);
+    replaceOpWithBufferizedValues(rewriter, op, loadOp.getSource());
+    return success();
+  }
+};
+
+struct StoreToMemrefOpInterface
+    : public BufferizableOpInterface::ExternalModel<
+          StoreToMemrefOpInterface, IREE::Codegen::StoreToMemrefOp> {
+  bool bufferizesToMemoryRead(Operation *op, OpOperand &opOperand,
+                              const AnalysisState &state) const {
+    return true;
+  }
+
+  bool bufferizesToMemoryWrite(Operation *op, OpOperand &opOperand,
+                               const AnalysisState &state) const {
+    return false;
+  }
+
+  bufferization::AliasingValueList
+  getAliasingValues(Operation *op, OpOperand &opOperand,
+                    const AnalysisState &state) const {
+    return {};
+  }
+
+  LogicalResult bufferize(Operation *op, RewriterBase &rewriter,
+                          const BufferizationOptions &options) const {
+    auto storeOp = cast<IREE::Codegen::StoreToMemrefOp>(op);
+    auto maybeBuffer =
+        getBuffer(rewriter, storeOp->getOpOperand(0).get(), options);
+    if (failed(maybeBuffer))
+      return failure();
+    Value srcMemref = *maybeBuffer;
+
+    // If everything bufferized inplace, no copy is needed. We wrote to the
+    // target buffer already. The copy folds away in that case.
+    if (failed(options.createMemCpy(rewriter, storeOp->getLoc(), srcMemref,
```

Comment:
optional nit: It looks like the code is copied from `DispatchTensorStoreOpInterface`, I'd write `storeOp.getLoc()`.